### PR TITLE
Feat: Final Price Calculation

### PR DESCRIPTION
**Pull request by pritam-kr**
### [Feat: calculate final price](https://github.com/pritam-kr/Pearl/commit/1a7a00a7e675e1ea8ad8b9bac86a1b86f184b7fe)

- Now users can calculate the final price for their product.
 I implemented the following calculation as  mentioned below

- Total Quantity

- Total Price

- Delivery Charge

- Total MRP

- Total Discount

- GST

- Final Price

Users can get calculations based on this point.
and delivery charge is applied only if users purchase above 15000 rupees.
and the charge is 1000.

Screenshot (12)

### CODE REVIEW

**Review by srishti-maurya — APPROVED** [review]
Code looks clean, and the use of comments are very descriptive 👍

File: src/utils/maxMinPrice.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
  // getting max price in all products
 
- export const gettingMaxPrice = (productData) => Math.max(...productData.map((eachProduct) => eachProduct.price))
+ export const gettingMaxPrice = (productData) => Math.max(...productData.map((eachProduct) => eachProduct.currentPrice))
```
Comment (poornima-alugubelly) [inline src/utils/maxMinPrice.js:3]: osm oneliner 🤩

File: src/pages/MyCart/MyCart.jsx [hunk 2]
```diff
@@ -2,12 +2,44 @@ import React from "react";
...
+import { priceFormatter } from "../../utils/priceFormatter";
 
 const MyCart = () => {
   const {
     state: { cart },
   } = useCartContext();
 
+  //get total Product quantity
+  const getTotalQuantity = cart.reduce(productQuantity, 0);
+
```
Comment (poornima-alugubelly) [inline src/pages/MyCart/MyCart.jsx:15]: I didn't know u could break reduce into pieces like this nice!

**Review by poornima-alugubelly — APPROVED** [review]
Good modular code 👍

**Review by SukanyaSen530 — APPROVED** [review]

**Pull request merged**